src/fury/encoder/row_encode_trait.h

```diff
@@ -17,7 +17,8 @@
  * under the License.
  */
 
-#pragma once
+#ifndef __ROW_ENCODER_TRAIT_HEADER
+#define __ROW_ENCODER_TRAIT_HEADER
```

Comment:
I think MSVC supports `pragma once`.

https://learn.microsoft.com/en-us/cpp/preprocessor/once?view=msvc-170

Reply:
The msvc I use does not support `#pragma once`, and it is also VS2022(14.38.33130). Is it more versatile if we use this method?

Reply:
https://learn.microsoft.com/en-us/cpp/preprocessor/pragma-directives-and-the-pragma-keyword?view=msvc-170

from this page, we can see that it is supported.

Reply:
And also, here is a live example on godbolt: https://godbolt.org/z/38e35q1rx

>  Is it more versatile if we use this method?
It is not related to this PR. Feel free to open issue or a seperate PR for it.